differs/rpm_diff.go

```diff
@@ -0,0 +1,312 @@
+		return packages, err
+	}
+
+	exitCode, err := client.WaitContainer(container.ID)
+	if err != nil {
+		return packages, err
+	}
+
+	outBuf := new(bytes.Buffer)
+	errBuf := new(bytes.Buffer)
+	logOpts := godocker.LogsOptions{
+		Context:      context.Background(),
+		Container:    container.ID,
+		Stdout:       true,
+		Stderr:       true,
+		OutputStream: outBuf,
+		ErrorStream:  errBuf,
+	}
+
+	if err := client.Logs(logOpts); err != nil {
+		return packages, err
+	}
+
+	if exitCode != 0 {
+		return packages, fmt.Errorf("non-zero exit code %d: %s", exitCode, errBuf.String())
+	}
+
+	output := strings.Split(outBuf.String(), "\n")
+	return parsePackageData(output)
+}
+
+// parsePackageData parses the package data of each line in rpmOutput and
+// returns a map of packages.
+func parsePackageData(rpmOutput []string) (map[string]util.PackageInfo, error) {
+	packages := make(map[string]util.PackageInfo)
+
+	for _, output := range rpmOutput {
+		spl := strings.Split(output, "\t")
+		if len(spl) != 3 {
+			// ignore the empty (last) line
+			if output != "" {
+				logrus.Errorf("unexpected rpm-query output: '%s'", output)
+			}
+			continue
+		}
+		pkg := util.PackageInfo{}
+
+		var err error
+		pkg.Size, err = strconv.ParseInt(spl[2], 10, 64)
+		if err != nil {
+			return packages, fmt.Errorf("error converting package size: %s", spl[2])
+		}
+
+		pkg.Version = spl[1]
+		packages[spl[0]] = pkg
+	}
+
+	return packages, nil
+}
+
+// loadArchiveToDaemon loads the image specified by archive to the docker
+// Daemon.
+func loadArchiveToDaemon(archive string, client *godocker.Client) (string, error) {
+	// read and write tar archive to input stream/buffer
+	inBuf := new(bytes.Buffer)
+	if content, err := ioutil.ReadFile(archive); err != nil {
+		return "", err
+	} else {
+		inBuf.Write(content)
+	}
+
+	outBuf := new(bytes.Buffer)
+	loadOpts := godocker.LoadImageOptions{
+		InputStream:  inBuf,
+		OutputStream: outBuf,
+		Context:      context.Background(),
+	}
+	if err := client.LoadImage(loadOpts); err != nil {
+		return "", err
+	}
+
+	// output has the form "Loaded image: repository:tag"
+	output := outBuf.String()
+	spl := strings.Split(output, ": ")
+	if len(spl) != 2 {
+		return "", fmt.Errorf("unexpected docker load output: %s", output)
+	}
+	image := strings.TrimSpace(spl[1])
+
+	return image, nil
+}
+
+// removeRepoTagsFromManifest removes all RepoTags from the manifest.json in
+// order to generate random image names when being loaded by the daemon.
+// Otherwise, docker would rename old images in case of name conflicts, which
+// is undesired behaviour from a user's perspective.
+func removeRepoTagsFromManifest(r io.Reader) ([]byte, error) {
+	f, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, err
+	}
+
+	var manifestItems []dockertar.ManifestItem
+	if err := json.Unmarshal(f, &manifestItems); err != nil {
+		return nil, fmt.Errorf("error unmarshalling manifest: %s", err)
+	}
+	for i := range manifestItems {
```

Comment:
Should we error here instead of the archive contained multiple images? It's sort of a corner-case, but the behavior will be a bit weird.

Reply:
Yes, we should do that. Thanks!